src/preprocess_info.cc

```diff
@@ -158,7 +195,236 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
+    }
+  } else {
+    return false;
+  }
+}
+
+void PreProcessInfo::SetPreProcessChannel(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 2) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[1].IsObject()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  size_t index = info[0].ToNumber().Int32Value();
+  Napi::Object newChannel = info[1].ToObject();
+
+  ie::PreProcessInfo& preInfo = _input_info->getPreProcess();
+
+  size_t numberOfChannels = preInfo.getNumberOfChannels();
+
+  if (numberOfChannels == 0) {
+    Napi::Error::New(env, "accessing pre-process when nothing was set.")
+        .ThrowAsJavaScriptException();
+    return;
+  };
+
+  if (index >= numberOfChannels || index < 0) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return;
+  };
+
+  ie::PreProcessChannel::Ptr channel = preInfo[index];
+  if (newChannel.Has("stdScale")) {
+    if (newChannel.Get("stdScale").IsNumber()) {
+      channel->stdScale = newChannel.Get("stdScale").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong stdScale").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanValue")) {
+    if (newChannel.Get("meanValue").IsNumber()) {
+      channel->meanValue = newChannel.Get("meanValue").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong meanValue").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanData")) {
+    if (!newChannel.Get("meanData").IsObject()) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object meanData = newChannel.Get("meanData").ToObject();
+
+    if (!meanData.Has("desc") || !meanData.Has("data")) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("desc").IsObject()) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    if (!meanData.Get("data").IsArrayBuffer()) {
+      Napi::TypeError::New(env, "Wrong data").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::Object desc = meanData.Get("desc").ToObject();
+    if (!checkDesc(desc)) {
+      Napi::TypeError::New(env, "Wrong desc").ThrowAsJavaScriptException();
+      return;
+    }
+
+    Napi::ArrayBuffer data =
+        Napi::ArrayBuffer::ArrayBuffer(env, meanData.Get("data"));
+    std::string precision_string = desc.Get("precision").ToString();
+    std::string layout_string = desc.Get("layout").ToString();
```

Comment:
You should validate the `desc` has `precision` and `layout` members and whether they are strings.

Reply:
You should also validate the content of `precision_string` and `layout_string` by `utils::IsValidLayoutName` and `utils::IsValidPrecisionName`.

Reply:
It has been completed by the function `checkDesc()`. Please look at here 
[Usage of checkDesc() ](https://github.com/lionkunonly/inference-engine-node-1/blob/db3d400795351f6610d357bb60d9d2b9cd3348c6/src/preprocess_info.cc#L368) and the implementation is here [Implementation of checkDesc() ](https://github.com/lionkunonly/inference-engine-node-1/blob/db3d400795351f6610d357bb60d9d2b9cd3348c6/src/preprocess_info.cc#L266).

Reply:
Do I need to move the statements in the `checkDesc()` here ?

Reply:
That's good.